### ISSUE TITLE
How to get parameters

### ISSUE THREAD

**Issue by meijuanwang** [Raymw/Federated-XGBoost#3]
Hi,
   I have a few questions about this project. I have read the paper, in the local node ,we need return the parameters. I didn't find the code of geting parameters. Did I miss something?

**Comment by Raymw** [comment]
Hi,
I use the Rabit for parameter passing which is built in XGBoost distributed mode, and I think this is more efficient than doing the parameter passing from scratch. However, in that paper, I aggregated those Gradients and Hessians locally for each client. So when communicating information between clients about best split point and the value of Gain during the process of training, only the aggregated value will be transmitted for calculating the Gain.
Thanks !

**Comment by meijuanwang** [comment]
Hi,
  Thank you for your answer. As I ubderstand it , the sever can get the passing parameter through Rabit, when rank=1, does it mean the server get new parameters? Sorry, I don't know much about Rabit.
Thanks

**Comment by Raymw** [comment]
Yes but the communication part has been implemented inside the XGBoost. Cuz the Rabit is invested by the group of XGBoost and they have done a good optimization in the distributed mode. You can see more details if you check the code of the original XGBoost package.
Thanks !

**Comment by meijuanwang** [comment]
Hi,
  In "readme" ,you mentioned  set two nodes (num-workers =2) locally to simulate the experiments, if I use more than two nodes , I need to set two situations: rank=0 and rank>0 . Is that right? Have you tried this experiments in two machines?
Thanks!

**Comment by Raymw** [comment]
yeh you can try more nodes and this is similar to MPI. And you can either reference rabit or MPI document for more details.
